compiler/src/org.graalvm.compiler.asm.aarch64/src/org/graalvm/compiler/asm/aarch64/AArch64Assembler.java

```diff
@@ -3002,19 +3004,73 @@ public void dc(DataCacheOperationType type, Register src) {
         emitInt(DC.encoding | type.encoding() | rt(src));
     }
 
+    /**
+     * Instruction Synchronization Barrier.
+     */
+    public void isb() {
+        emitInt(ISB.encoding);
+    }
+
     public void annotatePatchingImmediate(int pos, Instruction instruction, int operandSizeBits, int offsetBits, int shift) {
         if (codePatchingAnnotationConsumer != null) {
             codePatchingAnnotationConsumer.accept(new SingleInstructionAnnotation(pos, instruction, operandSizeBits, offsetBits, shift));
         }
     }
 
-    void annotateImmediateMovSequence(int pos, int numInstrs) {
-        if (codePatchingAnnotationConsumer != null) {
-            codePatchingAnnotationConsumer.accept(new MovSequenceAnnotation(pos, numInstrs));
+    public static class PatchableCodeAnnotation extends CodeAnnotation {
+        protected PatchableCodeAnnotation(int instructionStartPosition) {
+            super(instructionStartPosition);
+        }
+
+        @SuppressWarnings("unused")
+        void patch(int codePos, int relative, byte[] code) {
```

Comment:
Could this be an `abstract` method instead of throwing `shouldNotReachHere`?

Reply:
Done.